lib/inets/src/http_server/httpd.erl

```diff
@@ -49,6 +49,39 @@
 -deprecated({parse_query, 1,
             "use uri_string:dissect_query/1 instead"}).
 
+%%%========================================================================
+%%% Types
+%%%========================================================================
+-type property() :: atom().
+-type ets_table() :: ets:tid().
+
+%%%========================================================================
+%%% Callbacks
+%%%========================================================================
+-callback do(ModData) -> {proceed, OldData} | {proceed, NewData} | {break, NewData} | done when
+      ModData :: [{data,NewData} | {'Body', Body} | {'Head',Head}],
+      OldData :: list(),
+      NewData :: [{response, {StatusCode, Body}}],
+      StatusCode :: integer(),
+      Body :: iolist() | nobody | {Fun, FunArg},
+      Head :: [HeaderOption],
+      HeaderOption :: {Option, Value} | {code, StatusCode},
+      Option :: accept_ranges | allow,
+      Value :: string(),
+      FunArg :: [term()],
+      Fun :: fun((FunArg) -> sent | close | Body).
+
+-callback remove(ConfigDB) -> ok | {error, Reason} when
+      ConfigDB :: ets_table(), Reason :: term().
+
+-callback store({Option, Value}, Config) ->
+    {ok, {Option, NewValue}} | {error, Reason} when
+      Option :: property(),
+      Config :: [{Option, Value}],
+      Value :: term(),
+      NewValue :: term(),
+      Reason :: term().
+
```

Comment:
I think remove and store should be optional, as for instance mod_get only implements do!